src/vs/workbench/browser/treeController.ts

```diff
@@ -0,0 +1,35 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import { DefaultController, WorkbenchOpenMode, IControllerOptions } from 'vs/base/parts/tree/browser/treeDefaults';
+import { IConfigurationService } from 'vs/platform/configuration/common/configuration';
+
+interface IConfiguration {
+	workbench: {
+		openMode: string;
+	};
+}
+
+export class TreeControllerBase extends DefaultController {
+	constructor(
+		options: IControllerOptions,
```

Comment:
made it optional so child controllers don't need to pass in IControllerOptions if not needed